docs/vm-installation.md

```diff
@@ -0,0 +1,159 @@
+# Antrea Agent installation on VM
+
+Antrea Agent can run on a Linux or Windows VM, and enforce Antrea NetworkPolicies on the VM. This document describes
+the steps needed to configure and run `antrea-agent` on VMs.
+
+## Prerequisites on Kubernetes cluster
+
+1. Enable `ExternalNode` feature on the `antrea-controller`.
+2. Create a NameSpace for `antrea-agent`. This document will use `vm-ns` as an example NameSpace for illustration.
+
+```bash
+kubectl create ns vm-ns
+```
+
+3. Create a ServiceAccount, ClusterRole and ClusterRoleBinding for `antrea-agent` as shown below. If you use a different
+   Namespace other than `vm-ns`, you need to update the [VM RBAC manifest](../build/yamls/externalnode/vm-agent-rbac.yml)
+   and change `vm-ns` to the right Namespace.
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/antrea-io/antrea/feature/externalnode/build/yamls/externalnode/vm-agent-rbac.yml
+```
+
+4. Create `antrea-agent.kubeconfig` file for `antrea-agent` to access the K8S API server.
+
+```bash
+export CLUSTER_NAME="kubernetes"
+export SERVICE_ACCOUNT="vm-agent"
+APISERVER=$(kubectl config view -o jsonpath="{.clusters[?(@.name==\"$CLUSTER_NAME\")].cluster.server}")
+TOKEN=$(kubectl -n vm-ns get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='$SERVICE_ACCOUNT')].data.token}"|base64 --decode)
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-cluster kubernetes --server=$APISERVER --insecure-skip-tls-verify=true
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-credentials antrea-agent --token=$TOKEN
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-context antrea-agent@kubernetes --cluster=kubernetes --user=antrea-agent
+kubectl config --kubeconfig=antrea-agent.kubeconfig use-context antrea-agent@kubernetes
+# Copy antrea-agent.kubeconfig to the VM
+```
+
+5. Create `antrea-agent.antrea.kubeconfig` file for `antrea-agent` to access the `antrea-controller` API server.
+
+```bash
+# Specify the antrea-controller API server endpoint. Antrea-Controller needs to be exposed via the Node IP or a
+# public IP that is reachable from the VM
+export ANTREA_API_SERVER="https://172.18.0.1:443"
```

Comment:
Why not define a new varible for antrea cluster?

Reply:
Do you think instead of 
export ANTREA_API_SERVER="https://172.18.0.1:443", 
it should be something ANTREA_CLUSTER?

Reply:
I feel ANTREA_API_SERVER works better

Reply:
> Do you think instead of export ANTREA_API_SERVER="https://172.18.0.1:443", it should be something ANTREA_CLUSTER?

No, I mean since we have `export CLUSTER_NAME="kubernetes"`, why not having `export ANTREA_CLUSTER_NAME="antrea"`, because I see you have this command `kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-cluster antrea --server=$ANTREA_API_SERVER --insecure-skip-tls-verify=true`, in which "antrea" is hard-coded.

It is strong opinion.

Reply:
Make sense. Updated it.